sklearn/decomposition/tests/test_nmf.py

```diff
@@ -576,3 +591,16 @@ def test_nmf_custom_init_dtype_error():
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
         non_negative_factorization(X, H=H, update_H=False)
+
+
+# FIXME : should be removed in 0.26
+def test_init_default_deprecation():
+    # Test FutureWarning on init default
+    msg = ("The 'init' value, when 'init=None' and "
+           "n_components is less than n_samples and "
+           "n_features, will be changed from 'nndsvd' to "
+           "'nndsvda' in 0.26.")
+    rng = np.random.mtrand.RandomState(42)
+    A = np.abs(rng.randn(6, 5))
+    with pytest.warns(FutureWarning, match=msg):
+        nmf._initialize_nmf(A, 3)
```

Comment:
`_initialize_nmf` is private, we should test that there is a warning also in the public API, `non_negative_matrix_factorization` and `NMF().fit`. Actually, since we did not change the default parameter to "warn" in the public API, I think there is currently no warning.

Reply:
You are right. Fixed.

Reply:
We should do it in `non_negative_factorization` since it is public API as well.

Reply:
Right... I'm sorry, clearly it took me sometime to understand , hope next time I will be more efficient.

Reply:
No worry, that's why we do reviews !